src/main/scala/com/actian/spark_vector/colbuffer/string/IntegerEncodedStringColumnBuffer.scala

```diff
@@ -31,6 +32,36 @@ private[colbuffer] abstract class IntegerEncodedStringColumnBuffer(maxValueCount
   protected def encode(str: String): Int
 }
 
-private object IntegerEncodedStringColumnBuffer {
+private class ConstantLengthSingleByteStringColumnBuffer(p: ColumnBufferBuildParams) extends IntegerEncodedStringColumnBuffer(p) {
+
+  override protected def encode(value: String): Int = if (StringConversion.truncateToUTF8Bytes(value, 1).length == 0) {
+    IntegerEncodedStringColumnBuffer.Whitespace
+  } else {
+    value.codePointAt(0)
+  }
+}
+
+private class ConstantLengthSingleCharStringColumnBuffer(p: ColumnBufferBuildParams) extends IntegerEncodedStringColumnBuffer(p) {
+
+  override protected def encode(value: String): Int = if (Character.isHighSurrogate(value.charAt(0))) {
+    IntegerEncodedStringColumnBuffer.Whitespace
+  } else {
+    value.codePointAt(0)
+  }
+}
+
+/** Builds a `ColumnBuffer` object for `char`, `nchar` integer-encoded types. */
+private[colbuffer] object IntegerEncodedStringColumnBuffer extends ColumnBufferBuilder {
   final val Whitespace = '\u0020'
+
+  private val buildPartial: PartialFunction[ColumnBufferBuildParams, ColumnBufferBuildParams] = {
+    case p if p.precision == 1 => p
```

Comment:
It's a bit longer and inconsistent with p.precision > 0 (or 1 in some cases) from other buildPartial functions, like the ByteEncodedString's equivalent. Would've helped if there were more eq checks, e.g. p.precision == 1, p.scale == 100, p.name = "name", etc